.changeset/config.json

```diff
@@ -0,0 +1,7 @@
+{
+  "$schema": "https://unpkg.com/@changesets/config@0.2.1/schema.json",
+  "changelog": "@changesets/cli/changelog",
+  "commit": true,
+  "linked": [["@redux-saga/core", "redux-saga"]],
+  "access": "public"
```

Comment:
can `access` be configured differently for particular packages? I have other repo which mixes private & public packages and I'm wondering if such use case is already covered or if the support for it could be added (I could possibly work on adding it). 

Also - could the default just query `package.json` for a particular package? there is `publishConfig` field in `package.json` that could do a trick. The bot could also validate if that field is added correctly.

Reply:
Quick q: Do you mean packages that are `private: true` (not published at all) or ones that are `access: restricted` (privately published package). The first we support (we respect private: true), there's a bit more on the second below.

---

Technically, you can use publishConfig to do this, but there's behaviour with scoped packages where it assumes `--access=private`, which if your repo-level config is `public` you generally don't want.

I made an issue around ensuring you can have booth public and restricted packages in a repository:

https://github.com/atlassian/changesets/issues/193

Reply:
> Quick q: Do you mean packages that are private: true (not published at all) or ones that are access: restricted (privately published package). The first we support (we respect private: true), there's a bit more on the second below.

I meant here the latter (privately published) - but packages not published at all are also of my interest, so this is good to know.

> Technically, you can use publishConfig to do this, but there's behaviour with scoped packages where it assumes --access=private, which if your repo-level config is public you generally don't want.

Per package `publishConfig` could alleviate the issue altogether, right? As the decision would be local to a package and explicit.

Reply:
> Per package publishConfig could alleviate the issue altogether

I think we should respect per-package publish, but having a global config for when it's unspecified, mostly because of scoped packages. Most people using scoped packages publicly just do an initial publish with `--access="public"` and don't specify it in the config. When a new package is added later on, this often gets missed, causing initial release pain (Michael Jackson was complaining about this last week).

Letting the repository owner get it 'right' once is pretty powerful because it lowers the needed knowledge to someone contributing a new package.

We should definitely respect per-package config though 😅

Reply:
Gotcha 👌 would it be possible to add validation to changeset bot so it verifies if there is either per package or global `publishConfig` missing?

Reply:
Definitely. To the bot, or to the action? Or to the cli?

Reply:
(I would probably vote for the GH action that manages publishing being the place to add it, but it's possible it should just be part of the core library)

Reply:
> Definitely. To the bot, or to the action? Or to the cli?

I don't quite understand the difference between the bot and the action here - like bot "just" wraps the action, right? 

Without thinking about it too much I would vote for the CLI because it's always valuable - no matter if using the bot or not. Bot would only have to handle this CLI error to report it back nicely on GitHub.